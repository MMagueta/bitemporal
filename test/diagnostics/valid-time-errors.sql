\set QUIET 'on'
\pset footer off

drop schema if exists vt cascade;
create schema vt;

\echo 'bitemporal.valid_time_errors - [ok]'
\pset title 'vt.ok'
create table vt.ok( valid_period int4range not null );
select * from bitemporal.valid_time_errors('vt.ok');

\echo 'bitemporal.valid_time_errors - [missing valid time]'
\pset title 'vt.missing'
create table vt.missing( id int );
select * from bitemporal.valid_time_errors('vt.missing');

\echo 'bitemporal.valid_time_errors - [nullable]'
\pset title 'vt.nullable'
create table vt.nullable( valid_period int4range );
select * from bitemporal.valid_time_errors('vt.nullable');

\echo 'bitemporal.valid_time_errors - [invalid type]'
\pset title 'vt.invalid_type'
create table vt.invalid_type( valid_period int not null );
select * from bitemporal.valid_time_errors('vt.invalid_type');

\echo 'bitemporal.valid_time_errors - [nullable invalid type]'
\pset title 'vt.nullable_invalid_type'
create table vt.nullable_invalid_type( valid_period int );
select * from bitemporal.valid_time_errors('vt.nullable_invalid_type');
