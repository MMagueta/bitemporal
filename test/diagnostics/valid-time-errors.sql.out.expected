             vt.ok
 namespace | relation | message 
-----------+----------+---------

                       vt.missing
 namespace | relation |             message              
-----------+----------+----------------------------------
 vt        | missing  | Missing attribute "valid_period"

                              vt.nullable
 namespace | relation |                     message                     
-----------+----------+-------------------------------------------------
 vt        | nullable | Attribute "valid_period" should be not nullable

                                                vt.invalid_type
 namespace |   relation   |                                       message                                       
-----------+--------------+-------------------------------------------------------------------------------------
 vt        | invalid_type | Attribute "valid_period" with invalid type. Expected: "int4range", found: "integer"

                                                vt.nullable_invalid_type
 namespace |       relation        |                                       message                                       
-----------+-----------------------+-------------------------------------------------------------------------------------
 vt        | nullable_invalid_type | Attribute "valid_period" should be not nullable
 vt        | nullable_invalid_type | Attribute "valid_period" with invalid type. Expected: "int4range", found: "integer"

